src/sparseml/keras/models/external/keras_applications.py

```diff
@@ -0,0 +1,125 @@
+            continue
+        key = "keras_applications.{}".format(model_func_name)
+        input_shape = (
+            (224, 224, 3) if model_func_name != "InceptionV3" else (299, 299, 3)
+        )
+        arch, sub_arch = _get_architecture(model_func_name)
+
+        # wrap model constructor for registry compatibility
+        wrapped_constructor = _registry_constructor_wrapper(key, model_func)
+
+        ModelRegistry.register_wrapped_model_constructor(
+            wrapped_constructor,
+            key=key,
+            input_shape=input_shape,
+            domain="cv",
+            sub_domain="classification",
+            architecture=arch,
+            sub_architecture=sub_arch,
+            default_dataset="imagenet",
+            default_desc="base",
+            repo_source="keras.applications",
+        )
+
+
+def _registry_constructor_wrapper(key, model_func):
+    # wraps the keras_applications model constructor function to be compatible
+    # with sparseml model registry loading
+    def wrapper(
+        pretrained_path: str = None,
+        pretrained: Union[bool, str] = False,
+        pretrained_dataset: str = None,
+        **kwargs,
+    ):
+        """
+        :param pretrained_path: A path to the pretrained weights to load,
+            if provided will override the pretrained param
+        :param pretrained: True to load the default pretrained weights,
+            a string to load a specific pretrained weight
+            (ex: base, pruned-moderate),
+            or False to not load any pretrained weights
+        :param pretrained_dataset: The dataset to load pretrained weights for
+            (ex: imagenet, mnist, etc).
+            If not supplied will default to the one preconfigured for the model.
+        """
+        if isinstance(pretrained, str):
+            if pretrained.lower() == "true":
+                pretrained = True
+            elif pretrained.lower() in ["false", "none"]:
+                pretrained = False
+
+        weights = None
+        if pretrained:
+            if pretrained_dataset == "imagenet":
+                weights = "imagenet"  # Imagenet pretrained weights
+            elif pretrained_path is not None:
+                weights = pretrained_path  # Path to a weight file
+        if weights is not None:
+            _LOGGER.info("Model being created with weights from {}".format(weights))
+        else:
+            _LOGGER.info("Model being created with random weights")
+        try:
+            model = model_func(weights=weights, **kwargs)
+        except ValueError:
+            # Load the model directly assuming "weights" contain all information
+            # about the model
+            _LOGGER.info("Loading model directly from {}".format(weights))
+            model = keras.models.load_model(weights)
+        return model
+
+    return wrapper
+
+
+def _get_architecture(model_name):
+    if model_name == "ResNet50":
+        return "ResNet_v1", "50"
```

Comment:
This should be lower case in the model registry / sparsezoo

Reply:
Good catch, thanks! This was intended for pulling the pretrained model from keras apps then fine-tuning for imagenette (before pruning), but since we ended up not pushing the fine-tuning recipe (containing only LR mods) to the zoo, this code was not tested. The training script supports this use case though.